text/0037-anr-rates.md

```diff
@@ -0,0 +1,107 @@
+* Start Date: 2022-10-15
+* RFC Type: decision
+* RFC PR: [#37](https://github.com/getsentry/rfcs/pull/37)
+* RFC Status: draft
+
+# Summary
+
+Calculate ANR (App Not Responding) rates with session backed data to match Google Play apps Android vitals as closely as possible.
+
+# Motivation
+
+ANR rate is an app health diagnostic (much like crash rate) that Google Play provides to allow developers to compare stability of releases. Google Play Console (platform that Google provides for developers to monitor the performance of their app, where they would find ANR rate) is not accessible to all developers on a project/team. We want to surface ANR rates (and potentially user-perceived ANR rates) in Sentry to consolidate these app health diagnostics in one place.
+
+# Background
+
+ANRs are triggered if the UI thread on an Android app is blocked for too long. Android vitals provides the following [definition](https://developer.android.com/topic/performance/vitals/anr#android-vitals) for ANRs:
+
+**ANR rate:** The percentage of your daily active users who experienced any type of ANR.
+
+**User-perceived ANR rate:** The percentage of your daily active users who experienced at least one *user-perceived ANR*. Currently only ANRs of type `Input dispatching timed out` are considered user-perceived.
+
+**Daily active user**: Unique user who uses your app on a single day on a single device, potentially over multiple sessions. If a user uses your app on more than one device in a single day, each device will contribute to the number of active users for that day. If multiple users use the same device in a single day, this is counted as one active user.
+
+<aside>
+💡 User-perceived ANR rate is a *core vital* meaning that it affects the discoverability of your app on Google Play.
+</aside>
+
+
+ANRs are currently surfaced as error events with tag `mechanism:ANR`. With the data we already have on hand, we can calculate ANR rate as follows:
+
+```bash
+(unique users who experienced error events with tag mechanism: ANR)
+___________________________________________________________________ x 100 
+                    (total unique users)
+```
+
+There are a couple problems with this:
+
+1. Total ANRs is affected by client side sampling and dropped events if org/project is close to error quota
+2. The most accurate count of total unique users will probably come from the sessions dataset, or maybe even the metrics dataset (but again this one is affected by client-side sampling)
```

Comment:
i don't think we have any features for client-side sampling for sessions

Reply:
Ah yeah, by "metrics" here I meant the processed transactions/generic metrics - that would still be affected by client-side sampling. although come thing of it, can we even get unique users from that data? in any case yeah, the sessions data will have what we need

Reply:
Sessions don't do client-side sampling but error events do (such as an ANR event), so a user could technically experience 3 ANRs but maybe only 1 one of them got sampled.
If we want a more accurate rate, maybe we should expand the session metadata.